src/components/demos/read/Demo.js

```diff
@@ -195,12 +194,18 @@ export default function ReadDemo(props) {
             </DemoText>
           )}
         </Box>
-        {readDemo.renderModal &&
-          Modal({
-            update,
-            readDemo,
-            ProgressBar: (props) => <ProgressBar {...props} />,
-          })}
+        {state.renderModal && (
+          <Modal
+            type={'readDemo'}
+            update={update}
+            {...{
+              ...state,
+              ProgressBar: ({ update, readDemo }) => (
+                <ProgressBar update={update} readDemo={readDemo} />
+              ),
+            }}
```

Comment:
I honestly don't know what this does. Can we simplify this syntactically to reduce the amount of spreading? It looks likes it could be:

```
          <Modal
            type={'readDemo'}
            update={update}
            ProgressBar={({ update, readDemo }) => (
              <ProgressBar update={update} readDemo={readDemo} />
            )},
            {...state}
          />
```